src/lcals-kokkos/DIFF_PREDICT-Kokkos.cpp

```diff
@@ -0,0 +1,80 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DIFF_PREDICT.hpp"
+
+#if defined(RUN_KOKKOS)
+#include "common/KokkosViewUtils.hpp"
+#include <iostream>
+
+namespace rajaperf {
+namespace lcals {
+
+//
+// Kokkos helper function
+//
+
+template <class px_type, class cx_type>
+void diff_predict_helper(Index_type run_reps, Index_type ibegin,
```

Comment:
Is the plan to move to these helper methods more generally so you can use the macro loop bodies?

Reply:
@MrBurmark -- We (meaning David Poliakoff and me) created this helper function in the early days of our efforts to develop Kokkos variants.  I can rework this implementation to look like the rest -- I hear where you're coming from.